edx_event_bus_kafka/internal/consumer.py

```diff
@@ -282,11 +287,88 @@ def record_event_consuming_error(self, run_context, error, maybe_event):
             # and will only read the exception from stack context.
             raise Exception(error)
         except BaseException:
+            self._add_message_monitoring(run_context=run_context, message=maybe_event, error=error)
             record_exception()
             logger.exception(
                 f"Error consuming event from Kafka: {error!r} in context {context_msg} -- {event_msg}"
             )
 
+    def _add_message_monitoring(self, run_context, message, error=None):
+        """
+        Record additional details for monitoring.
+
+        Arguments:
+            run_context: Dictionary of contextual information: full_topic, consumer_group,
+              and expected_signal.
+            message: None if event could not be fetched or decoded, or a Message if one
+              was successfully deserialized but could not be processed for some reason
+            error: (Optional) An exception instance, or None if no error.
+        """
+        try:
+            kafka_message, kafka_error = self._get_kafka_message_and_error(message=message, error=error)
+
+            # .. custom_attribute_name: kafka_topic
```

Comment:
**Note to reviewer:** Feel free to comment on anything around these attributes, including names or whether you want to see different attributes, etc.
[Mark resolved when done reading.]

Reply:
I think it might be worthwhile to add event_type

Reply:
Done.